py/torch_tensorrt/dynamo/converter_registry.py

```diff
@@ -0,0 +1,327 @@
+        # Dict[Target, Callable] or Dict[Target, Sequence[ConverterSupport]]
+        # Where, for the latter, the sequence must be non-empty
+        for registry in self.registries:
+            for converters in registry.values():
+                if isinstance(converters, (list, tuple)):
+                    assert (
+                        all(isinstance(c, ConverterSupport) for c in converters)
+                        and len(converters) > 0
+                    )
+                else:
+                    assert callable(converters), "Converter function must be callable"
+
+    def __getitem_without_validation__(self, key: Target):
+        """Get the first-found converter in any registry
+
+        Searches all registries in order and returns the first converter encountered
+        """
+        if isinstance(key, Node):
+            raise KeyError(
+                "Unvalidated accesses to the Converter registry can only be "
+                + "made with node targets. Try accessing the registry with node.target"
+            )
+
+        self.validate_invariants()
+
+        # Iterate over all registries and return the first converter found
+        for registry in self.registries:
+            if key in registry:
+                converters = registry[key]
+
+                if isinstance(converters, (list, tuple)):
+                    return converters[0].converter_implementation
+                else:
+                    return converters
+
+        raise KeyError(f"None of the converter registries have an entry for {key}")
+
+    def __getitem__(self, node: Node):
+        """Get the first-found validated converter in any registry
+
+        Searches all registries in order and returns the first converter
+        which passes validation on the input node
+        """
+        if not isinstance(node, Node):
+            raise KeyError(
+                "Validated accesses to the Converter registry can only be "
+                + "made with node inputs. Try accessing the registry with a node "
+                + "or use get_unvalidated to access without node validation."
+            )
+
+        self.validate_invariants()
+        key = node.target
+
+        # Iterate over all registries, validating the converter on the input node
+        # If no capability_validator function is found, assume full coverage
+        for registry in self.registries:
+            if key in registry:
+                converters = registry[key]
+
+                if isinstance(converters, (list, tuple)):
+                    for candidate in converters:
+                        if candidate.capability_validator(node):
+                            return candidate.converter_implementation
+                else:
+                    return converters
+
+        raise KeyError(
+            f"None of the converter registries have a validated entry for {key}, with node {node}"
+        )
```

Comment:
`__getitem__` and `get` access the registry with `Node` inputs only, while `__getitem_without_validation__` and `get_unvalidated` allow access with `Target` inputs only.

`__contains__` dispatches the access based on the type of input. `Node` inputs call a validated implementation while `Target` inputs call an unvalidated implementation.